src/panels/CreateClusterPanel.ts

```diff
@@ -162,11 +164,32 @@ async function createCluster(
         errorMessage: null,
     });
 
-    const clusterSpec = getManagedClusterSpec(location, name);
+    // kubernetes version is required to create a cluster via deployments
+    const kubernetesVersion = await containerServiceClient.managedClusters.listKubernetesVersions(location);
+    if (!kubernetesVersion || !kubernetesVersion.values || kubernetesVersion.values.length === 0 || !kubernetesVersion.values[0].version) {
+        window.showErrorMessage(`No Kubernetes versions available for location ${location}`);
+        webview.postProgressUpdate({
+            event: ProgressEventType.Failed,
+            operationDescription,
+            errorMessage: "No Kubernetes versions available for location",
+        });
+        return;
+    }
+
+    const clusterSpec: ClusterSpec = {
+        location,
+        name,
+        resourceGroupName: group.name,
+        subscriptionId: subscriptionContext.subscriptionId,
+        kubernetesVersion: kubernetesVersion.values[0].version, // selecting the latest version since versions come in descending order
+        username: subscriptionContext.userId
+    };
+
+    const deploymentSpec = getManagedClusterSpec(clusterSpec, preset);
 
     try {
-        const poller = await containerServiceClient.managedClusters.beginCreateOrUpdate(group.name, name, clusterSpec);
-        poller.onProgress((state) => {
+        const poller = await resourceManagementClient.deployments.beginCreateOrUpdate(group.name, name, deploymentSpec);
```

Comment:
We should probably [generate a unique name](https://learn.microsoft.com/en-us/azure/azure-resource-manager/templates/deploy-cli#azure-deployment-template-name) for the deployment, rather than using the cluster name, to avoid earlier deployments getting overwritten by later ones.

This will probably only be a problem when things go wrong and users try to create the same cluster a second time...but those are precisely the times when users are likely to want a consistent deployment history.